### ISSUE TITLE
declare_interior_mutable_const triggered in tokio::task_local

### ISSUE THREAD

**Issue by rukai** [rust-lang/rust-clippy#9224]
### Summary

@kyoto7250 this might be of interest to you given you solved a similar issue before.

as of tokio 1.20.0, `tokio::task_local!` triggers this lint.

I thought that https://github.com/rust-lang/rust-clippy/pull/9015 might have resolved the issue but it should be in the latest nightly and the issue is still occurring in the latest nightly.

### Lint Name

declare_interior_mutable_const

### Reproducer

I tried this code:

```rust
tokio::task_local! {
    pub static CONTEXT_CHAIN_NAME: String;
}
```

I saw this happen:

```
warning: a `const` item should never be interior mutable
   --> shotover-proxy/src/transforms/mod.rs:373:1
    |
373 | / tokio::task_local! {
374 | |     pub static CONTEXT_CHAIN_NAME: String;
375 | | }
    | |_^
    |
    = note: `#[warn(clippy::declare_interior_mutable_const)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#declare_interior_mutable_const
    = note: this warning originates in the macro `$crate::__thread_local_inner` which comes from the expansion of the macro `tokio::task_local` (in Nightly builds, run with -Z macro-backtrace for more info)

```

I expected to see this happen:
not trigger the lint, I expect tokio to have a sound implementation, if it doesnt then we should raise an issue on the tokio repo instead.

### Version

```text
rustc 1.64.0-nightly (d68e7ebc3 2022-07-20)
binary: rustc
commit-hash: d68e7ebc38cb42b8b237392b28045edeec761503
commit-date: 2022-07-20
host: x86_64-unknown-linux-gnu
release: 1.64.0-nightly
LLVM version: 14.0.6
```
